### PR TITLE
Remove all usage of the GitHub API

### PR DESCRIPTION
**Pull request by cmcgee1024**
The GitHub API was previously used to retrieve previous release nightly snapshot toolchains. There's no reason for someone to use swift.org to get access to those toolchains. Also, removing GitHub API dependencies from swiftly cleans up the code, and removes the test mock responses for the tags.

Remove the remains of the GitHub API from the code, tests, and documentation.

### CODE REVIEW

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Comment by cmcgee1024** [comment]
macOS is working in spite of reporting a CI failure:

```
Test Suite 'All tests' passed at 2024-10-19 14:42:19.909.
	 Executed 66 tests, with 0 failures (0 unexpected) in 82.744 (82.752) seconds
```

**Comment by kkebo** [comment]
As far as I can tell from reading the logs, the actual error is this:

```
swiftpm_testing_helper/Entrypoint.swift:29: Fatal error: Failed to open test bundle at path /error: Exited with unexpected signal code 5
```

The error occurred in [swiftpm-testing-helper](https://github.com/swiftlang/swift-package-manager/blob/4095b908832f25c492af72728dc6687924978604/Sources/swiftpm-testing-helper/Entrypoint.swift#L29).

And the following code is related:

- https://github.com/swiftlang/swift-package-manager/blob/4095b908832f25c492af72728dc6687924978604/Sources/PackageModel/UserToolchain.swift#L399
- https://github.com/swiftlang/swift-package-manager/blob/4095b908832f25c492af72728dc6687924978604/Sources/Commands/SwiftTestCommand.swift#L952

On Darwin, SwiftPM seems to run swiftpm-testing-helper for swift-testing tests. SwiftPM runs swift-testing tests after XCTest tests by default even if there are no swift-testing test cases.

So the test results looked like:

- XCTest tests
  - All tests passed
- swift-testing
  - **Crashed**

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

File: Sources/Swiftly/ListAvailable.swift [hunk 1]
```diff
@@ -51,11 +51,13 @@ struct ListAvailable: SwiftlyCommand {
 
         switch selector {
         case let .snapshot(branch, _):
-            if case let .release(major, _) = branch, major < 6 {
-                throw Error(message: "Listing available snapshots previous to 6.0 is not supported.")
+            do {
+                tc = try await SwiftlyCore.httpClient.getSnapshotToolchains(platform: config.platform, branch: branch).map { ToolchainVersion.snapshot($0) }
+            } catch let branchNotFoundError as SwiftlyHTTPClient.SnapshotBranchNotFoundError {
+                throw Error(message: "The snapshot branch \(branchNotFoundError.branch) was not found on swift.org. Note that snapshot toolchains are only available for the current `main` release and the previous x.y (major.minor) release.")
```
Comment (justice-adams-apple) [inline Sources/Swiftly/ListAvailable.swift:57]: Might want to use the word `latest` here as opposed to `previous`. In the update command we say

>""The branch \(branchNotFoundErr.branch) doesn't have any snapshots available on swift.org so this snapshot build cannot be updated. It is possible that there has been a new release on swift.org and the previous release snapshot are no longer available. Install a fresh snapshot toolchain from the either the latest release x.y (major.minor) or from the main branch.""

Where we tell users "previous" are gone, install "latest". But here we say that main and "previous" are available.

Basically just some very small confusion around previous != latest

File: Sources/Swiftly/Update.swift [hunk 2]
```diff
@@ -195,9 +195,18 @@ struct Update: SwiftlyCommand {
...
-            return try await SwiftlyCore.httpClient.getSnapshotToolchains(platform: config.platform, branch: old.branch, limit: 1) { snapshot in
-                snapshot.branch == old.branch && snapshot.date > old.date
-            }.first.map(ToolchainVersion.snapshot)
+            let newerSnapshotToolchains: [ToolchainVersion.Snapshot]
+            do {
+                newerSnapshotToolchains = try await SwiftlyCore.httpClient.getSnapshotToolchains(platform: config.platform, branch: old.branch, limit: 1) { snapshot in
+                    snapshot.branch == old.branch && snapshot.date > old.date
+                }
+            } catch let branchNotFoundErr as SwiftlyHTTPClient.SnapshotBranchNotFoundError {
+                throw Error(message: "The branch \(branchNotFoundErr.branch) doesn't have any snapshots available on swift.org so this snapshot build cannot be updated. It is possible that there has been a new release on swift.org and the previous release snapshot are no longer available. Install a fresh snapshot toolchain from the either the latest release x.y (major.minor) or from the main branch.")
```
Comment (justice-adams-apple) [inline Sources/Swiftly/Update.swift:204]:
```suggestion
                throw Error(message: "The branch \(branchNotFoundErr.branch) doesn't have any snapshots available on swift.org so this snapshot build cannot be updated. It is possible that there has been a new release on swift.org and the previous release snapshots are no longer available. Install a fresh snapshot toolchain from the either the latest release x.y (major.minor) or from the main branch.")
```

**Review by justice-adams-apple — APPROVED** [review]
Super small typos, otherwise LGTM

**Comment by cmcgee1024** [comment]
@swift-ci test macOS

**Comment by cmcgee1024** [comment]
macOS CI tests are all passing despite reporting an error here.

**Pull request merged**